test/development/typescript-plugin/metadata/warn-no-type.test.ts

```diff
@@ -0,0 +1,134 @@
+import type { PluginLanguageService } from '../test-utils'
+import {
+  getPluginLanguageService,
+  getTsFiles,
+  NEXT_TS_ERRORS,
+} from '../test-utils'
+import { join } from 'path'
+
+const fixturesDir = join(__dirname, 'app/warn-no-type')
+
+describe('typescript-plugin - metadata - warn-no-type', () => {
+  let languageService: PluginLanguageService
+
+  beforeAll(() => {
+    languageService = getPluginLanguageService(__dirname)
+  })
+
+  it('should not have diagnostics for metadata with type', () => {
+    const hasTypeDir = join(fixturesDir, 'metadata', 'has-type')
+    const tsFiles = getTsFiles(hasTypeDir)
+    expect(tsFiles.length).toBe(8)
+
+    const totalDiagnostics = []
+    for (const tsFile of tsFiles) {
+      const diagnostics = languageService.getSemanticDiagnostics(tsFile)
+      if (diagnostics.length > 0) {
+        console.warn(
+          `${tsFile}\n\nExpected 0 diagnostic but received ${diagnostics.length}.`
+        )
```

Comment:
This should be in the assertion. Otherwise the failure message and the warning may not be aligned by the test runner.